### PR TITLE
Adding Navi10 macropad

### PR DESCRIPTION
**Pull request by emdarcher**
Adding the Navi10 macropad to QMK.

## Description
Adding my new macropad, the Navi10, to QMK. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/navi10/keymaps/default/config.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 Ethan Durrant (emdarcher)
```
Comment (fauxpark) [inline keyboards/navi10/keymaps/default/config.h]: This file can be deleted.

File: keyboards/navi10/keymaps/emdarcher/config.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 Ethan Durrant (emdarcher)
```
Comment (fauxpark) [inline keyboards/navi10/keymaps/emdarcher/config.h]: This file can be deleted.

File: keyboards/navi10/readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Navi10 / ナビ10
+
+![navi10](https://i.imgur.com/QpFCMFd.jpg)
+
+A simple and fun navigation cluster macropad with exposed components.
+
+Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)
```
Comment (fauxpark) [inline keyboards/navi10/readme.md]:
```suggestion
* Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)
```

File: keyboards/navi10/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Navi10 / ナビ10
+
+![navi10](https://i.imgur.com/QpFCMFd.jpg)
+
+A simple and fun navigation cluster macropad with exposed components.
+
+Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)  
+Hardware Supported: Navi10
```
Comment (fauxpark) [inline keyboards/navi10/readme.md]:
```suggestion
* Hardware Supported: Navi10
```

File: keyboards/navi10/readme.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Navi10 / ナビ10
+
+![navi10](https://i.imgur.com/QpFCMFd.jpg)
+
+A simple and fun navigation cluster macropad with exposed components.
+
+Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)  
+Hardware Supported: Navi10  
+Hardware Availability: [KeyHive](https://www.keyhive.xyz/)
```
Comment (fauxpark) [inline keyboards/navi10/readme.md]:
```suggestion
* Hardware Availability: [KeyHive](https://www.keyhive.xyz/)
```

File: keyboards/navi10/rev0/rules.mk [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]: Please remove this section.

File: keyboards/navi10/rev0/rules.mk [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
```

File: keyboards/navi10/rev0/rules.mk [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: keyboards/navi10/rev0/rules.mk [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes       # Audio control and System control
```

File: keyboards/navi10/rev0/rules.mk [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+CONSOLE_ENABLE = yes        # Console for debug(+400)
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]:
```suggestion
CONSOLE_ENABLE = yes        # Console for debug
```

File: keyboards/navi10/rev0/rules.mk [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]:
```suggestion
MIDI_ENABLE = no            # MIDI support
```

File: keyboards/navi10/rev0/rules.mk [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (fauxpark) [inline keyboards/navi10/rev0/rules.mk]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

File: keyboards/navi10/rev2/rules.mk [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+TAP_DANCE_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/navi10/rev2/rules.mk]: As above.

File: keyboards/navi10/rev3/rules.mk [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = yes           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+TAP_DANCE_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/navi10/rev3/rules.mk]: As above.

File: keyboards/navi10/readme.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
+# Navi10 / ナビ10
+
+![navi10](https://i.imgur.com/QpFCMFd.jpg)
+
+A simple and fun navigation cluster macropad with exposed components.
+
+* Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)
```
Comment (fauxpark) [inline keyboards/navi10/readme.md]:
```suggestion
* Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)
```

File: keyboards/navi10/readme.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
+# Navi10 / ナビ10
+
+![navi10](https://i.imgur.com/QpFCMFd.jpg)
+
+A simple and fun navigation cluster macropad with exposed components.
+
+* Keyboard Maintainer: [emdarcher](https://github.com/emdarcher)  
+* Hardware Supported: Navi10
```
Comment (fauxpark) [inline keyboards/navi10/readme.md]:
```suggestion
* Hardware Supported: Navi10
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**